src/index.js

```diff
@@ -0,0 +1,158 @@
+// Add state
+const state = {
+  temp: 32,
+};
+
+// **********************************************
+// Temperature Increase and Decrease Buttons
+// **********************************************
+
+// update state (raise temp) function
+// Fahrenheit***
+const updateTemp = () => {
+  const tempElement = document.querySelector('#tempContainer');
+  tempElement.textContent = `${state.temp}°F`;
+};
+
+// Make a function to run when event occurs (behavior).
+// Fahrenheit***
+const increaseTemp = () => {
+  const tempContainer = document.querySelector('#tempContainer');
+  const tempSpan = document.createElement('span');
+  tempContainer.appendChild(tempSpan);
+  state.temp += 1;
+  updateTemp();
+};
+
+const decreaseTemp = () => {
+  const tempContainer = document.querySelector('#tempContainer');
+  const tempSpan = document.createElement('span');
+  tempContainer.appendChild(tempSpan);
+  state.temp -= 1;
+  updateTemp();
+};
+
+// **********************************************
+// Temperature Changing Number Text Color
+// **********************************************
+
+const colorChange = () => {
+  if (state.temp > 79) {
+    document.querySelector('#tempContainer').style.color = '#dd0a35';
+  } else if (state.temp > 69) {
+    document.querySelector('#tempContainer').style.color = '#f87d42';
+  } else if (state.temp > 59) {
+    document.querySelector('#tempContainer').style.color = '#ffce3e';
+  } else if (state.temp > 49) {
+    document.querySelector('#tempContainer').style.color = '#b9e937';
+  } else {
+    document.querySelector('#tempContainer').style.color = '#bae8e8';
+  }
+};
```

Comment:
Consider the above logic and how similar all the branches are. We look for whether the temperature is within a range, and pick a color accordingly. What if we had a list of records that we could iterate through to find the values. We could set up something like
```js
  const tempRecords = [
    { lowerBound: 79, color: "#dd0a35"},
    { lowerBound: 69, color: "other color"},
    ...
  ];
```

For the `else` case, we might set the `lowerBound` to a very low value, or to `null` and write a special case to check it. But for everything else, we could simply iterate through the list and find the first record that has a lower bound _lower_  than our temperature, then use it as the source of picking the color!

We could take a similar approach for the landscape.